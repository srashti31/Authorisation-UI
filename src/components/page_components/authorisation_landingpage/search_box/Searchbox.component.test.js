import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SearchBox from "./Searchbox.component";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

describe("search box Component", () => {
    it("should match snapshot", () => {
        const snapshot = renderer.create(<SearchBox />).toJSON();
        expect(snapshot).toMatchSnapshot();
    });
    it("should render without crashing", () => {
        shallow(<SearchBox />);
    });
    it("should simulate clear button ", () => {
        const wrapper = shallow(
            <SearchBox />
        );
        wrapper.find(".clear-search").at(0).simulate("click", {});
        expect(wrapper.find(".search-input").at(0).props().value).toEqual("");

    });
});
