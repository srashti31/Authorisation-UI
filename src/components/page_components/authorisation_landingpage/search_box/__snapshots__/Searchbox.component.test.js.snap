// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search box Component should match snapshot 1`] = `
<div
  className="search-contain"
>
  <div
    className="search-box"
  >
    <input
      className="search-input"
      id="search"
      placeholder="Enter a customer’s ID…"
      value="83468484"
    />
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-root-1 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </span>
    </button>
  </div>
  <button
    className="clear-search"
    onClick={[Function]}
  >
    RESET SEARCH
  </button>
</div>
`;
